gulpfile.js

```diff
@@ -40,16 +51,127 @@
     );
 
     gulp.task(
-        'less',
+        'build:clean',
+        function (callback) {
+            rimraf('public/css', callback);
+        }
+    );
+
+    gulp.task(
+        'build:less',
         function () {
             return gulp
                 .src('public/less/main.less')
-                .pipe(less({ paths: [ __dirname + '/public/less' ] }))
+                .pipe(less({
+                    paths: [ 'public/less' ]
+                }))
                 .pipe(gulp.dest('public/css'));
         }
     );
 
     gulp.task(
+        'build',
+        [
+            'build:clean',
+            'build:less'
+        ]
+    );
+
+    gulp.task(
+        'package:clean',
+        function (callback) {
+            if (environment === 'development') {
+                throw new Error('Cannot use "package" tasks in development environment');
+            }
+            rimraf('dist/' + environment, function () {
+                mkdirp('dist/' + environment, callback);
+            });
+        }
+    );
+
+    gulp.task(
+        'package:less',
+        function () {
+            if (environment === 'development') {
+                throw new Error('Cannot use "package" tasks in development environment');
+            }
+            return gulp
+                .src('public/less/main.less')
+                .pipe(less({
+                    paths: [ 'public/less' ],
+                    optimize: true
+                }))
+                .pipe(gulp.dest('dist/' + environment));
+        }
+    );
+
+    gulp.task(
+        'package:javascript',
+        function () {
+            if (environment === 'development') {
+                throw new Error('Cannot use "package" tasks in development environment');
+            }
+            // Dynamically generate the `include` list, because most of the modules are dynamically `require`d.
+            recursive('public/app', function (err, files) {
+                return gulp
+                    .src('public/app/main.js')
+                    .pipe(requirejs({
+                        mainConfigFile: 'public/app/main.js',
+                        name: '../lib/requirejs/require',
+                        optimize: 'none',
```

Comment:
See call to `uglify()` below -- we can't do it here because then the `replace` on line 129 won't work, because the variable name, quoting style, semicolon presence, etc, will be unknown.